### PR TITLE
Feature/add the different components

### PR DESCRIPTION
**Pull request by Bluette1**
#### Feature/add the different components - Milestone 2
- Prepare the directory structure
- Create a components directory in the src folder

- Implement the App component to render the Display and ButtonPanel component
 - Wrap the child components inside `React.Fragment`, as this let you group the children's components without adding extra nodes to the DOM

- Implement the Display component to show the result of the calculation, with `0` as the default

- Implement the Button component to show the button name

- Implement the ButtonPanel component
    -Render the calculator panel with all the buttons in the following groups: (use divs to divide buttons into groups)

 Result: A "pretty ugly page full of non-functional buttons". Styling will be catered for in a later milestone.

[Live Demo](https://calculator-lite.herokuapp.com/)

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Button = props => {
```
Comment (Forison) [inline src/components/Button.js:4]: **[OPTIONAL]**
-You could also destructure at the argument level like this `const Button = ({result}) => {}`

File: src/helpers/buttonGroups.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const buttons = [
+  { itemList: ['AC', '+/-', '%', '÷'], key: 'group-1' },
```
Comment (Forison) [inline src/helpers/buttonGroups.js:2]: - We can refactor this to simply like this
`[
  { itemList1: ['AC', '+/-', '%', '÷'] },  
  { itemList2: ['7', '8', '9', 'x'] },
  { itemList3: ['4', '5', '6', '-'] },
  { itemList4: ['0', '.', '=']},
]`;

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Changes required❌
# 🙋‍♂️ Marylene,
#### You have done a great job so far, however, there are a few changes we have to make to get this project approved🤗.
`App` component
* [ ] App.js has children components wrapped in [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax) short syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/))
- fragment are preferred because they are fast and no extra node is created.
### That is it kindly make the changes and submit a re-review 👨‍🔧.
#### Happy coding👨‍💻 
Addo Forison

File: src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import '../App.css';
+import ButtonGroups from './ButtonGroups';
+import buttonGroups from '../helpers/buttonGroups';
+import Display from './Display';
+
+const App = () => (
+  <div className="App">
```
Comment (Forison) [inline src/components/App.js:7]: Kindly get rid of `<div className="App">`, it defeats the purpose of using fragment in the first place. the component should just look like this 
`
 <>
      <Display />
      <ButtonGroups groups={buttonGroups()} />
    </>
`

File: src/components/ButtonGroups.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ButtonPanel from './ButtonPanel';
+
+const ButtonGroups = props => {
+  const { groups } = props;
+
+  return (
+    <div>
```
Comment (Forison) [inline src/components/ButtonGroups.js:9]: Kindly remove this `<div>` as well it defeats the purpose of using fragment in the first place.

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::change required ❌
#### Good job on addressing some of the issues raised in the previous review, I am sorry if I wasn't clear enough with the recommendation on the use of fragments **[Please [read this article](https://dev.to/tumee/react-fragments-what-why-how-2kh1) before you make the changes I asked]**
- kindly check my comment below on how to implement this requirement😇.
### That is it kindly make the change and submit a re-review👨‍🔧
#### Happy coding👨‍💻
Addo Forison

**Comment by Bluette1** [comment]
@Forison, Okay, thank you very much for the review!

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @Bluette1 :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Comment by Bluette1** [comment]
@muriloroque, thank you very much for the review!

**Pull request merged**